include/bsp/iotc_bsp_io_net.h

```diff
@@ -18,73 +18,88 @@
+ * iotc_bsp_mem_alloc() | Allocates memory and returns a pointer to the allocated block. |
+ * iotc_bsp_mem_free() | Frees a block of memory. |
+ * iotc_bsp_mem_realloc() | Changes the size of a memory block and returns a pointer to the reallocated block. |
+ * iotc_bsp_rng_get() | Generates and returns a random, 32-bit integer. |
+ * iotc_bsp_rng_init() | Initializes the platform-specific RNG requirements. |
+ * iotc_bsp_rng_shutdown() | Shuts down the RNG and frees all of the resources from initializing and generating random numbers. |
+ * iotc_bsp_sha256() | Generates a SHA256 cryptographic hash. |
+ * iotc_bsp_time_getcurrenttime_milliseconds() | Gets the milliseconds since Epoch. |
+ * iotc_bsp_time_getcurrenttime_seconds() | Gets the seconds since Epoch. |
+ * iotc_bsp_time_getmonotonictime_milliseconds() | Gets the monotonic time in milliseconds. |
+ * iotc_bsp_time_init() | Initializes the platform-specific timekeeping requirements. |
+ * iotc_bsp_tls_cleanup() | Frees a TLS context from memory and deletes any associated data. |
+ * iotc_bsp_tls_connect() | Starts a TLS handshake. |
+ * iotc_bsp_tls_init() | Initializes a TLS library and creates a TLS context. |
+ * iotc_bsp_tls_pending() | Gets the pending readable bytes. |
+ * iotc_bsp_tls_read() | Decrypts MQTT messages. |
+ * iotc_bsp_tls_write() | Encrypts MQTT messages. |
+ *
+ * # POSIX BSP
+ * The POSIX BSP is in the
+ * <code><a href="~/src/bsp/platforms/posix">src/bsp/platforms/posix</a></code>
+ * directory. If you're customizing the BSP for a new device, you can refer
+ * to this implementation as an example.
+ *
+ * # TLS implementations
+ * The SDK has turn-key
+ * <a href="../../../src/bsp/tls/mbedtls">mbedTLS</a>
+ * and <a href="~/src/bsp/tls/wolfssl">wolfSSL</a> implementations. The 
+ * default <code>make</code> target
+ * <a href="../../../src/bsp/tls/mbedtls">downloads and builds</a> mbedTLS.
+ *
+ * To use the turn-key wolfTLS implementation, specify 
+ * <code>IOTC_BSP_TLS=wolfssl</code> in the
+ * <a href="../../../doc/porting_guide.md#build-environment">main makefile</a>.
+ *
+ * The turn-key TLS libraries consist of customized
+ * <a href="dd/d79/iotc__bsp__tls_8h.html">iot_bsp_tls.h</a> and
+ * <a href="dd/d79/iotc__bsp__crypto_8h.html">iotc_bsp_crypto.h</a>
+ * functions:
+ * | mbedTLS | wolfSSL |
+ * | --- | --- |
+ * | <a href="../../../src/bsp/tls/mbedtls/iotc_bsp_tls_mbedtls.c">src/bsp/tls/mbedtls/iotc_bsp_tls_mbedtls.c</a> | <a href="../../../src/bsp/tls/wolfssl/iotc_bsp_tls_wolfssl.c">src/bsp/tls/wolfssl/iotc_bsp_tls_wolfssl.c</a> |
+ * | <a href="../../../src/bsp/crypto/mbedtls/iotc_bsp_crypto.c">src/bsp/crypto/mbedtls/iotc_bsp_crypto.c</a> | <a href="../../../src/bsp/crypto/wolfssl/iotc_bsp_crypto.c">src/bsp/crypto/wolfssl/iotc_bsp_crypto.c</a> |
  */
 
 /**
  * @file  iotc_bsp_io_net.h
- * @brief Perform asynchronous networking.
+ * @brief Creates and manages sockets in the native socket library.
```

Comment:
I miss here the async expression, it is an important message to the implementers. Majority of the socket use cases are sync (blocking) socket. For the SDK to work a socket must be async (non-blocking).

Reply:
native socket library --> device / board specific native socket library

Reply:
Good call - WDYT about "Creates and manages asynchronous sockets in the device's native socket library"

Reply:
👍 perfectly phrased!!